include/stl2/detail/algorithm/is_sorted.hpp

```diff
@@ -22,27 +22,23 @@
 // is_sorted [is.sorted]
 //
 STL2_OPEN_NAMESPACE {
-	template<ForwardIterator I, Sentinel<I> S, class Comp = less,
-		class Proj = identity>
-	requires
-		IndirectStrictWeakOrder<
-			Comp, projected<I, Proj>>
-	bool is_sorted(I first, S last, Comp comp = {}, Proj proj = {})
-	{
-		return last == __stl2::is_sorted_until(std::move(first), last,
-			std::ref(comp), std::ref(proj));
-	}
+	struct __is_sorted_fn : private __niebloid {
+		template<ForwardIterator I, Sentinel<I> S, class Proj = identity,
+			IndirectStrictWeakOrder<projected<I, Proj>> Comp = less>
+		constexpr bool operator()(I first, S last, Comp comp = {}, Proj proj = {}) const
+		{
+			return last == is_sorted_until(std::move(first), last, std::ref(comp), std::ref(proj));
```

Comment:
`std::ref` won't work in a `constexpr` function AFAIK

Reply:
True. The `constexpr invoke` proposal will fix it. I suppose we should make these `move`s (when possible) or copies (when not) for the time being.

Reply:
Better solution: leave these as-is, I'll implementing the parts of [P1065](https://wg21.link/p1065) that we need.

Reply:
Resolved by cjdb/cmcstl2#21.